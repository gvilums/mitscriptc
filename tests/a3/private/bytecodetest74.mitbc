function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [None],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [print, x, y, z],
			instructions = 
			[
				load_global	0
				load_global	1
				call	1
				pop
				load_global	0
				load_global	2
				call	1
				pop
				load_global	0
				load_global	3
				call	1
				pop
				load_const	0
				return
			]
		}
	],
	constants = [8],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [x, y, print, input, intcast],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	2
		load_func	1
		alloc_closure	0
		store_global	3
		load_func	2
		alloc_closure	0
		store_global	4
		load_const	0
		store_global	0
		load_func	3
		alloc_closure	0
		store_global	1
		load_global	1
		call	0
		pop
	]
}