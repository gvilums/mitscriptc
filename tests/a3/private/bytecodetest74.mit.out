8
FUNCTION
UninitializedVariableException: z
